add custom code feature for stream connectors (#119)
centreon/centreon-stream-connector-scripts: modules/docs/custom_code.md

1. stream connector init (only done on cbd reload or restart)
2. filter events
3. format event
4. put event in a queue
5. send all events stored in the queue

The second step has a set of filters but they may not be enough for you. This is where a custom code file can be useful.

Let's keep our idea of filtering events with hosts that do not have **notes** and see what it will looks like with real code

```lua
local self = ...

if not self.event.cache.host.notes or self.event.cache.host.notes == "" then
  -- the boolean part of the return is here to tell the stream connector to ignore the event
  return self, false
end

-- if the host has a note then we let the stream connector continue his work on this event
return self, true
-- do not forget 
```

## Use all the above chapters

### Add methods from other modules

What if we start logging what our custom code does? To do so, we can use [the warning method](sc_logger.md#warning-method)

```lua
local self = ...

if not self.event.cache.host.notes or self.event.cache.host.notes == "" then

>   -- use the warning method of oru logger module 

Reviewer: ```suggestion
  -- use the warning method of from the logger module 
```

  self.sc_logger:warning("[custom_code]: host: "
    .. tostring(self.event.cache.host.name) .. " do not have notes, therefore, we drop the event")
  -- the boolean part of the return is here to tell the stream connector to ignore the event
  return self, false
end

-- if the host has a note then we let the stream connector continue his work on this event
return self, true
-- new line after true
```

Maybe you want a closer look at what is inside the `self.event` table. To do so, we can dump it in our logfile using [the Dumper method](sc_common.md#dumper-method)

```lua
local self = ...

-- we dump the event table to have a closer look to all the available data from the event itself 
-- and all the things that are in the cache that we may want to use
self.sc_logger:notice("[custom_code]: self.event table data: " .. self.sc_common:dumper(self.event))

if not self.event.cache.host.notes or self.event.cache.host.notes == "" then

>   -- use the warning method of oru logger module 

Reviewer: ```suggestion
  -- use the warning method from the logger module 
```
